### PR TITLE
Fix JHU Calculation 03/22-Current

### PR DESCRIPTION
**Pull request by crwilcox**
Later time ranges are formatted as counties. As we were using the last state format date, we were stopping soon.

### CODE REVIEW

File: libs/CovidDatasets.py [hunk 1]
```diff
@@ -196,14 +196,15 @@ def combine_state_county_data(self, country, state):
             [self.DATE_FIELD, self.COUNTRY_FIELD, self.STATE_FIELD], as_index=False
         )[[self.CASE_FIELD, self.DEATH_FIELD, self.RECOVERED_FIELD]].sum()
         # Now we fill in whatever gaps we can in the state data using the county data
-        curr_date = state_data[self.DATE_FIELD].max()  # Start on the last date of state data we have
+        curr_date = max(state_data[self.DATE_FIELD].max(), county_data[self.DATE_FIELD].max()) # Start on the last date of state data we have
```
Comment (tsebens) [inline libs/CovidDatasets.py]: This is an interesting thought, one I hadn't considered.

In general we try to use the state data as our first choice and fill in the holes with county data. I don't imagine that's likely for the county data goes later than the state data, but I think we should have a plan in place for when it does. 

I think that we should end when the state data ends, because we are projecting at a state level, and often times summing counties provides a lower total than the state data. In this highly hypothetical situation, it is likely that the final cycle or two of just county data would artificially drag down the model.

In other words, I don't think we should do this, just wanted to talk through why 😄

File: libs/CovidDatasets.py [hunk 2]
```diff
@@ -196,14 +196,15 @@ def combine_state_county_data(self, country, state):
...
         county_data_to_insert = []
         while curr_date > self._START_DATE:
             curr_date -= datetime.timedelta(days=1)
             # If there is no state data for a day, we need to get some country data for the day
             if len(state_data[state_data[self.DATE_FIELD] == curr_date]) == 0:
                 county_data_for_date = copy(county_data[county_data[self.DATE_FIELD] == curr_date])
                 if len(county_data_for_date) == 0:  # If there's no county data, we're SOL.
+                    logging.info("NO COUNTY DATA: {}".format(curr_date))
```
Comment (tsebens) [inline libs/CovidDatasets.py]: I think we should take this one step further and raise an exception. If we reach this point in execution, that means that for a certain date, there is no state data and no county data. That's a serious problem, which _will_ crash the model later on in execution, 100% of the time.

The exception should also indicate the date for which there is no state or county data

File: libs/CovidDatasets.py (hunk 1 above)
Comment (davidstrauss) [reply libs/CovidDatasets.py]: > summing counties provides a lower total than the state data

It also sometimes yields a higher one, as is the case in Washington state.

Comment (davidstrauss) [reply libs/CovidDatasets.py]: For JHU, though, I think each data set has exclusively been states or counties. I could be wrong, though.

File: libs/CovidDatasets.py (hunk 2 above)
Comment (davidstrauss) [reply libs/CovidDatasets.py]: I don't want to block merging on this, but I'll open an issue for how we handle this concern. I'm leaning toward not throwing an exception and having the consumer of the data be responsible for ensuring it has the necessary data points to operate.

Comment (davidstrauss) [reply libs/CovidDatasets.py]: The issue: https://github.com/covid-projections/covid-data-model/issues/35
